src/parcsr_mv/par_csr_matop.c

```diff
@@ -6695,39 +6695,39 @@ hypre_ParCSRMatrixBlockColSumHost( hypre_ParCSRMatrix     *A,
    /* Communication variables */
    hypre_ParCSRCommPkg    *comm_pkg          = hypre_ParCSRMatrixCommPkg(A);
    HYPRE_Int               num_sends         = hypre_ParCSRCommPkgNumSends(comm_pkg);
-   HYPRE_Int              *send_map_elmts    = hypre_ParCSRCommPkgSendMapElmts(comm_pkg);
-   HYPRE_Int              *send_map_starts   = hypre_ParCSRCommPkgSendMapStarts(comm_pkg);
+   HYPRE_Int              *send_map_elmts;
+   HYPRE_Int              *send_map_starts;
 #if defined(HYPRE_USING_PERSISTENT_COMM)
    hypre_ParCSRPersistentCommHandle *comm_handle;
 #else
    hypre_ParCSRCommHandle           *comm_handle;
 #endif
 
    /* Update commpkg offsets */
-   hypre_ParCSRCommPkgUpdateVecStarts(comm_pkg, num_cols_block_B, 1, num_cols_block_B);
+   hypre_ParCSRCommPkgUpdateVecStarts(comm_pkg, 1, 0, 1);
+   send_map_elmts  = hypre_ParCSRCommPkgSendMapElmts(comm_pkg);
+   send_map_starts = hypre_ParCSRCommPkgSendMapStarts(comm_pkg);
 
    /* Allocate the recv and send buffers  */
 #if defined(HYPRE_USING_PERSISTENT_COMM)
    comm_handle = hypre_ParCSRCommPkgGetPersistentCommHandle(HYPRE_COMM_PKG_JOB_COMPLEX, comm_pkg);
    recv_data = (HYPRE_Complex *) hypre_ParCSRCommHandleRecvDataBuffer(comm_handle);
    send_data = (HYPRE_Complex *) hypre_ParCSRCommHandleSendDataBuffer(comm_handle);
    send_data = hypre_Memset((void *) send_data, 0,
-                            (size_t) (num_cols_offd_A * num_cols_block_B) * sizeof(HYPRE_Complex),
```

Comment:
Were these allocations a bug before? Looks like the new allocation is smaller?

Reply:
Not a bug, but as for my previous comment, this change was due to a simplification of the previous implementation of this function